Upstream-Status: Backport

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From b026b6e0428f91570a403eaf3ec28f176c80b0af Mon Sep 17 00:00:00 2001
From: Daniel Bratell <bratell@opera.com>
Date: Wed, 3 Apr 2019 10:13:59 +0000
Subject: [PATCH 19/27] [ozone/wayland] Drop dependency wayland ->
 ui/display/manager

The dependency wayland -> ui/display/manager created a
dependency loop since ui/display/manager depends on chromeos
which depends on a lot of stuff, and with some upcoming dep
fixes, eventually back on ozone and wayland.

Change-Id: I675047cd724818c9a26c7160c62ea4d4d3fc4749
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/1549071
Commit-Queue: Daniel Bratell <bratell@opera.com>
Reviewed-by: Maksim Sisov <msisov@igalia.com>
Cr-Commit-Position: refs/heads/master@{#647209}
---
 ui/ozone/platform/wayland/BUILD.gn                  | 1 -
 ui/ozone/platform/wayland/ozone_platform_wayland.cc | 3 +--
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/ui/ozone/platform/wayland/BUILD.gn b/ui/ozone/platform/wayland/BUILD.gn
index 7981c682fae6..3d71dc1b3124 100644
--- a/ui/ozone/platform/wayland/BUILD.gn
+++ b/ui/ozone/platform/wayland/BUILD.gn
@@ -108,7 +108,6 @@ source_set("wayland") {
     "//ui/base",
     "//ui/base:buildflags",
     "//ui/base/ime/linux",
-    "//ui/display/manager",
     "//ui/events",
     "//ui/events:dom_keycode_converter",
     "//ui/events/keycodes:xkb",
diff --git a/ui/ozone/platform/wayland/ozone_platform_wayland.cc b/ui/ozone/platform/wayland/ozone_platform_wayland.cc
index c368863f8e5b..202e146bcbc6 100644
--- a/ui/ozone/platform/wayland/ozone_platform_wayland.cc
+++ b/ui/ozone/platform/wayland/ozone_platform_wayland.cc
@@ -13,7 +13,6 @@
 #include "base/memory/ptr_util.h"
 #include "ui/base/buildflags.h"
 #include "ui/base/cursor/ozone/bitmap_cursor_factory_ozone.h"
-#include "ui/display/manager/fake_display_delegate.h"
 #include "ui/events/ozone/layout/keyboard_layout_engine_manager.h"
 #include "ui/events/system_input_injector.h"
 #include "ui/gfx/linux/client_native_pixmap_dmabuf.h"
@@ -114,7 +113,7 @@ class OzonePlatformWayland : public OzonePlatform {
 
   std::unique_ptr<display::NativeDisplayDelegate> CreateNativeDisplayDelegate()
       override {
-    return std::make_unique<display::FakeDisplayDelegate>();
+    return nullptr;
   }
 
   std::unique_ptr<PlatformScreen> CreateScreen() override {
-- 
2.20.1

